src/Features/Core/Shared/Utilities/LinkedFilesSymbolEquivalenceComparer.cs

```diff
@@ -30,14 +30,14 @@ bool IEqualityComparer<ISymbol>.Equals(ISymbol x, ISymbol y)
         {
             return x.Kind == y.Kind && (x.IsKind(SymbolKind.Local) || x.IsKind(SymbolKind.Label) || x.IsKind(SymbolKind.RangeVariable))
                  ? x.Name == y.Name
-                 : _symbolEquivalenceComparer.Equals(x, y);
+                 : x.Name == y.Name && x.Kind == y.Kind && _symbolEquivalenceComparer.Equals(x.ContainingType, y.ContainingType);
```

Comment:
What happens if one document has an method named M(), and the other has an extension method named M() that applies to this receiver type, or if M() is in a different spot in the hierarchy.  I don't think we want to show the warning in those cases either, even though `ContainingType` will be different.  Can you add some tests for those cases too?